File: omv-server/src/main/java/omv/server/entities/Visualization.java
package omv.server.entities;

import java.util.ArrayList;

import org.semanticweb.owlapi.model.OWLOntology;

import io.vertx.core.json.JsonObject;

public class Visualization {
    public OwlInfo owl1;
    public OwlInfo owl2;
    public Mapping mapping;
    public String output;

    public void create (OWLOntology owl1Object, OWLOntology owl2Object, JsonObject mappingobject) {
        try {
            this.owl1 = new OwlInfo(owl1Object);
            this.owl2 = new OwlInfo(owl2Object);
            this.mapping = new Mapping();
            this.mapping.init(mappingobject);
            //this.markEntitiesByRule(1);
            //this.markEntitiesByRule(2);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void markEntitiesByRule(int owlnumber) {
        OwlInfo owl = (owlnumber == 1) ? this.owl1 : this.owl2;
        ArrayList<String> relations = new ArrayList<String>();
        relations.add("Equivalent");
        relations.add("LinkedWith");
        for (String relation : relations) {
            ArrayList<Rule> rules;
            if (relation.equals("Equivalent")) {
                rules = this.mapping.classrules;
            } else {
                rules = this.mapping.proprules;
            }
            for (Rule rule : rules) {
                RuleEntity rulentity = (owlnumber == 1) ? rule.entity1 : rule.entity2;
                for (Object classruleobject : rulentity.classes) {
                    JsonObject classrule = (JsonObject) classruleobject;
                    OwlClassNode foundentity = owl.owlclasses.findByIri(classrule.getString("iri"));
                    if (foundentity != null) {
                        if (relation.equals("Equivalent")) {
                            foundentity.hasEquivalentRule = true;
                        } else if (relation.equals("LinkedWith")) {
                            foundentity.hasLinkedWithRule = true;
                        } else {
                            foundentity.hasOtherRule = true;
                        }
                    }
                }
                for (Object objectpropruleobject : rulentity.objectprops) {
                    JsonObject objectproprule = (JsonObject) objectpropruleobject;
                    OwlObjectPropertyNode foundentity = owl.owlobjprops.findByIri(objectproprule.getString("iri"));
                    if (foundentity != null) {
                        if (relation.equals("Equivalent")) {
                            foundentity.hasEquivalentRule = true;
                        } else if (relation.equals("LinkedWith")) {
                            foundentity.hasLinkedWithRule = true;
                        } else {
                            foundentity.hasOtherRule = true;
                        }
                    }
                }
                for (Object objectpropruleobject : rulentity.dataprops) {
                    JsonObject dataproprule = (JsonObject) objectpropruleobject;
                    OwlDataPropertyNode foundentity = owl.owldataprops.findByIri(dataproprule.getString("iri"));
                    if (foundentity != null) {
                        if (relation.equals("Equivalent")) {
                            foundentity.hasEquivalentRule = true;
                        } else if (relation.equals("LinkedWith")) {
                            foundentity.hasLinkedWithRule = true;
                        } else {
                            foundentity.hasOtherRule = true;
                        }
                    }
                }
            }
        }
    }

    public JsonObject toJsonObject() {
        JsonObject result = new JsonObject();
        result.put("owl1", this.owl1.toJsonObject());
        result.put("owl2", this.owl2.toJsonObject());
        result.put("mapping", this.mapping.toJsonOjbect());
        return result;
    }
}

File: omv-client/src/entities/Visualization.ts
import { Mapping } from "./Mapping";
import { OwlInfo } from "./OwlInfo";
import { Statistics } from "./Statistics";

export class Visualization {
    owl1: OwlInfo;
    owl2: OwlInfo;
    mapping: Mapping;
    statistics: Statistics;

    /** @param {any} Visualizationobject
     * @return {Visualization}
    */
    static fromObject (Visualizationobject: any): Visualization {
        const visualization: Visualization = new Visualization();
        visualization.owl1 = OwlInfo.fromObject(Visualizationobject.owl1);
        visualization.owl2 = OwlInfo.fromObject(Visualizationobject.owl2);
        visualization.mapping = Mapping.fromObject(Visualizationobject.mapping, visualization);
        visualization.mapping.findOrderedRules(visualization.owl1, visualization.owl2);
        visualization.statistics = new Statistics();
        visualization.statistics.init(visualization);


        // calc total rules
        visualization.owl1.owlclasses.calcTotalRules();
        visualization.owl1.owlobjprops.calcTotalRules();
        visualization.owl1.owldataprops.calcTotalRules();
        visualization.owl2.owlclasses.calcTotalRules();
        visualization.owl2.owlobjprops.calcTotalRules();
        visualization.owl2.owldataprops.calcTotalRules();
        return visualization;
    }
}